src/tools/roc_recv/main.cpp

```diff
@@ -249,9 +251,12 @@ int main(int argc, char** argv) {
         return 1;
     }
 
-    sndio::Pump pump(
-        sample_buffer_pool, receiver, *sink, config.common.internal_frame_size,
-        args.oneshot_flag ? sndio::Pump::ModeOneshot : sndio::Pump::ModePermanent);
+    sndio::Pump pump(sample_buffer_pool, receiver, *sink,
+                     packet::timestamp_from_ns(config.common.internal_frame_size,
+                                               config.common.output_sample_rate)
+                         * packet::num_channels(config.common.output_channels),
+                     args.oneshot_flag ? sndio::Pump::ModeOneshot
+                                       : sndio::Pump::ModePermanent);
```

Comment:
As with roc_pipeline, could you move these calculations inside sndio::Pump constructor? And add validation as well.

Reply:
done.

Reply:
done.